index.js

```diff
@@ -439,6 +474,62 @@ class TuyaDevice extends EventEmitter {
     }
   }
 
+  /**
+   * Create a deferred promise that resolves as soon as the connection is established.
+   */
+  createDeferredConnectPromise() {
```

Comment:
I don't think we need this, can we just return `new Promise()` at the end?

Reply:
The problem is that the 3.4 "connect" is in the end 3 messages exchanged, sothe final "resolve" will come from out an other method but we also still want to have "connect timeout" logic. So in the end for a "clean" solution we in any case need to store a promise.

The alternative is that we resolve connect promise already after we sent out the "start session exchange message", but in fact we are not yet connected.

Reply:
Right, I agree we need to store a Promise, just not sure why we need to have this wrapper that doesn't seem to have any custom logic to create one.

Reply:
This wrapper is needed because you can not just store the two methods reject and resolve... I experimented around a long time on how to build a deferred promise and this solution is used in one of my other projects and originally came from an other developer that uses it in his projects.

In fact if you return a promise you have no real way to reference to the reject or resolve method of this object "outside" of the Promise constructor. Thats why this is the trick to allow this.

It cost me hours to come to this in my other project :-)

You find it also here e.g. https://www.getrevue.co/profile/masteringjs/issues/promises-and-the-deferred-antipattern-235313

Reply:
(and I decided against adding a dependency lib to get a Deffered option which is kind of just doing exactly also that - e.g. https://github.com/ljharb/promise-deferred/blob/main/index.js

Reply:
PS: If you mean that I should kill the extra function and directly integrate the code into the connect function then yes I can do this. ;-)

Reply:
Read the code again and I understand the issue now, seems ok for now.